src/storage/fileio/mod.rs

```diff
@@ -0,0 +1,167 @@
+use std::{
+    fs::{File, OpenOptions},
+    io,
+};
+
+#[cfg(target_family = "unix")]
+use std::os::unix::prelude::AsRawFd;
+
+#[cfg(target_family = "unix")]
+use std::os::unix::prelude::FileExt;
+
+pub struct FileIO {}
```

Comment:
I would rather us have a `trait FileIO` and implementations like `FileIOImplLinux`, etc.

Reply:
You can keep the trait in this file and spin off the implementations into their own files in this module.

Reply:
We don't necessarily have to pass out trait object pointers with vtables, we can inject specific structs for each platform with something like `T: FileIO`, this wouldn't have a runtime impact and should still allow us to abstract common functionality away.